File: sample/src/main/java/contacts/sample/cheatsheet/sim/java/UpdateSimContactsActivity.java
package contacts.sample.cheatsheet.sim.java;

import android.app.Activity;

import contacts.core.ContactsFactory;
import contacts.core.entities.*;
import contacts.core.sim.SimContactsUpdate;

public class UpdateSimContactsActivity extends Activity {

    SimContactsUpdate.Result updateSimContact(SimContact simContact) {
        MutableSimContact mutableSimContact = simContact.mutableCopy();
        mutableSimContact.setName("Vandolf");
        mutableSimContact.setNumber("1234567890");

        return ContactsFactory.create(this)
                .sim()
                .update()
                .simContact(simContact, mutableSimContact)
                .commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/sim/kotlin/UpdateSimContactsActivity.kt
package contacts.sample.cheatsheet.sim.kotlin

import android.app.Activity
import contacts.core.Contacts
import contacts.core.entities.SimContact
import contacts.core.sim.SimContactsUpdate

class UpdateSimContactsActivity : Activity() {

    fun updateSimContact(simContact: SimContact): SimContactsUpdate.Result = Contacts(this)
        .sim()
        .update()
        .simContact(simContact, simContact.mutableCopy {
            name = "Vandolf"
            number = "1234567890"
        })
        .commit()
}